src/solver/optimisation/constraints/AreaBalance.cpp

```diff
@@ -7,45 +7,43 @@ static void shortTermStorageBalance(const ::ShortTermStorage::AREA_INPUT& shortT
     {
         unsigned index = storage.clusterGlobalIndex;
         constraintBuilder.ShortTermStorageInjection(index, 1.0)
-        .ShortTermStorageWithdrawal(index, -1.0);
+          .ShortTermStorageWithdrawal(index, -1.0);
     }
 }
 
 void AreaBalance::add(int pdt, int pays)
 {
-    CORRESPONDANCES_DES_CONTRAINTES& CorrespondanceCntNativesCntOptim
-      = problemeHebdo->CorrespondanceCntNativesCntOptim[pdt];
-    CorrespondanceCntNativesCntOptim.NumeroDeContrainteDesBilansPays[pays]
-      = problemeHebdo->ProblemeAResoudre->NombreDeContraintes;
+    data.CorrespondanceCntNativesCntOptim[pdt].NumeroDeContrainteDesBilansPays[pays]
+      = builder.data.nombreDeContraintes;
 
-        ConstraintNamer namer(problemeHebdo->ProblemeAResoudre->NomDesContraintes);
-        namer.UpdateTimeStep(problemeHebdo->weekInTheYear * 168 + pdt);
-        namer.UpdateArea(problemeHebdo->NomsDesPays[pays]);
-        namer.AreaBalance(problemeHebdo->ProblemeAResoudre->NombreDeContraintes);
+    ConstraintNamer namer(builder.data.NomDesContraintes);
+    namer.UpdateTimeStep(builder.data.weekInTheYear * 168 + pdt);
+    namer.UpdateArea(builder.data.NomsDesPays[pays]);
+    namer.AreaBalance(builder.data.nombreDeContraintes);
 
-        builder.updateHourWithinWeek(pdt);
+    builder.updateHourWithinWeek(pdt);
 
-        int interco = problemeHebdo->IndexDebutIntercoOrigine[pays];
-        while (interco >= 0)
-        {
+    int interco = data.IndexDebutIntercoOrigine[pays];
+    while (interco >= 0)
+    {
         builder.NTCDirect(interco, 1.0);
-        interco = problemeHebdo->IndexSuivantIntercoOrigine[interco];
-        }
+        interco = data.IndexSuivantIntercoOrigine[interco];
+    }
 
-    interco = problemeHebdo->IndexDebutIntercoExtremite[pays];
+    interco = data.IndexDebutIntercoExtremite[pays];
     while (interco >= 0)
     {
         builder.NTCDirect(interco, -1.0);
-        interco = problemeHebdo->IndexSuivantIntercoExtremite[interco];
+        interco = data.IndexSuivantIntercoExtremite[interco];
     }
 
-    exportPaliers(*problemeHebdo, builder, pays);
+    ExportPaliers(data.PaliersThermiquesDuPays[pays], builder);
     builder.HydProd(pays, -1.0)
       .Pumping(pays, 1.0)
       .PositiveUnsuppliedEnergy(pays, -1.0)
       .NegativeUnsuppliedEnergy(pays, 1.0);
 
-    shortTermStorageBalance(problemeHebdo->ShortTermStorage[pays], builder);
+    shortTermStorageBalance(data.ShortTermStorage[pays], builder);
```

Comment:
Same remark for short term storage than for thermal cluster : 
`builder.ShortTermStorageBalance(pdt, pays);` ?